contrib/haskell-ci-regenerate.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
+
+# Change to project root directory
+# https://stackoverflow.com/a/246128 and https://stackoverflow.com/a/8426110
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+cd "$(dirname "$DIR")"
+
+# Regenerate .travis.yml
+haskell-ci regenerate
+
+# Ensure that the patch still applies cleanly
+patch --reverse --dry-run .travis.yml .travis.yml.patch
```

Comment:
https://github.com/haskell-CI/haskell-ci/issues/257

Reply:
I.e is there some missing functionality in `haskell-ci` regarding patch files?

Reply:
My bad, I misunderstood how the patch is applied. I thought this was a Travis thing, not a `haskell-ci` thing. I did wonder why `--reverse` was needed.

Sorry. So my new understanding is that `haskell-ci regenerate` will itself apply the patch - so if `haskell-ci regenerate` succeeded then we know that the patch was applied cleanly.

In that case this script is unnecessary, and I will remove it. Thanks very much @phadej!